kube-core/src/object.rs

```diff
@@ -21,7 +21,10 @@ pub struct ObjectList<T>
 where
     T: Clone,
 {
-    // NB: kind and apiVersion can be set here, but no need for it atm
+    /// The type fields, always present
```

Comment:
I hope that this is generally true now. I feel like there were cases where it wasn't always present, but you're doing a `#[serde(default)]` so that should be safe either way.

Reply:
Yes, it may be not fully compatible with older kubernetes versions (can’t say which one), but the serialization should handle it. Currently this info is always returned. The type meta is not populated on the objects in the `items` field, so it would be weird to have an ObjectList returned without any type meta whatsoever, so I think this is safe.